### PR TITLE
Update dep

### PR DESCRIPTION
**Pull request by julienje**

### CODE REVIEW

**Comment by julienje** [comment]
I don't have access to the result of the snapshot build. Can somebody help me to know what is the problem?

**Comment by yahavi** [comment]
> I don't have access to the result of the snapshot build. Can somebody help me to know what is the problem?

@julienje,
Sorry for the inconvenience. We plan to automatically add a comment to the PR with the build log if the execution failed.

For now, I'm attaching the build log manually:
```
[Info] Downloading bintray/jcenter/org/jfrog/buildinfo/build-info-extractor-gradle/4.18.0/build-info-extractor-gradle-4.18.0-uber.jar
[Info] Downloaded 1 artifact.
[Info] Running gradle command: ./gradlew --init-script /root/.jfrog/dependencies/gradle/4.18.0/gradle.init clean artifactoryPublish -x test
Downloading https://services.gradle.org/distributions/gradle-5.4.1-all.zip
..............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
Unzipping /root/.gradle/wrapper/dists/gradle-5.4.1-all/1jkdto6ec7o8pep4874j0q1o8h/gradle-5.4.1-all.zip to /root/.gradle/wrapper/dists/gradle-5.4.1-all/1jkdto6ec7o8pep4874j0q1o8h
Set executable permissions for: /root/.gradle/wrapper/dists/gradle-5.4.1-all/1jkdto6ec7o8pep4874j0q1o8h/gradle-5.4.1/bin/gradle

Welcome to Gradle 5.4.1!

Here are the highlights of this release:
 - Run builds with JDK12
 - New API for Incremental Tasks
 - Updates to native projects, including Swift 5 support

For more details see https://docs.gradle.org/5.4.1/release-notes.html


FAILURE: Build failed with an exception.

* Where:
Build file '/var/opt/jfrog/pipelines/data/build_java_client_snapshot/runs/1843/steps/Snapshot/8966/dependencyState/resources/javaClientSnapshotGit/build.gradle' line: 15

* What went wrong:
A problem occurred evaluating root project 'artifactory-java-client'.
> Failed to apply plugin [id 'com.jfrog.artifactory']
   > Cannot cast object 'org.jfrog.gradle.plugin.artifactory.dsl.ArtifactoryPluginConvention@735c0454' with class 'org.jfrog.gradle.plugin.artifactory.dsl.ArtifactoryPluginConvention' to class 'org.jfrog.gradle.plugin.artifactory.dsl.ArtifactoryPluginConvention'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 15s
[Error] exit status 1
```

**Comment by yahavi** [comment]
@julienje 
The root cause of this failure is an issue in our automatic tests. These tests were added a few days ago and they still not working as expected.
We will fix them shortly.

**Comment by julienje** [comment]
Hi @yahavi
Do you have some new?

File: api/src/main/java/org/jfrog/artifactory/client/model/RepositorySummary.java [hunk 1]
```diff
@@ -6,22 +6,79 @@
  * @author Aviad Shikloshi
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
-public interface RepositorySummary {
+public class RepositorySummary {
```
Comment (yahavi) [inline api/src/main/java/org/jfrog/artifactory/client/model/RepositorySummary.java:9]: What is the reason for changing this interface to class?

**Review by yahavi — COMMENTED** [review]
Thanks for your contribution, @julienje!
The code looks great. I ran the tests locally and they passed.
Is there any reason for changing RepositorySummary from interface to class?

File: api/src/main/java/org/jfrog/artifactory/client/model/RepositorySummary.java (hunk 1 above)
Comment (julienje) [reply api/src/main/java/org/jfrog/artifactory/client/model/RepositorySummary.java:9]: The serialisation / deserialisation don't work anymore with new version of third party library. The code in RepositorySummaryDeserialize with the TypeReference on a list doesn't compile anymore.
It's sure that jackson need to know a concrete type. I choose the same approach that have CopyMoveResultMessage and CopyMoveResultReport. The CopyMoveResultMessage  is not an interface.
Was my explanation good enough?
I understand the basic idea of interface for api but not for dto object.

Comment (yahavi) [reply api/src/main/java/org/jfrog/artifactory/client/model/RepositorySummary.java:9]: Yes. Thank you

**Review by yahavi — APPROVED** [review]
LGTM

**Comment by julienje** [comment]
Hi @yahavi 
Do you know if a release is planed?

**Pull request merged**